core/domain/story_domain.py

```diff
@@ -0,0 +1,369 @@
+    story.
+    """
+
+    def __init__(
+            self, node_id, destination_node_ids,
+            acquired_skill_ids, prerequisite_skill_ids,
+            outline, exploration_id):
+        """Initializes a StoryNode domain object.
+
+        Args:
+            node_id: str. The unique id for each node.
+            destination_node_ids: list(str). The list of destination node ids
+                that this node points to in the story graph.
+            acquired_skill_ids: list(str). The list of skill ids acquired by
+                the user on completion of the node.
+            prerequisite_skill_ids: list(str). The list of skill ids required
+                before starting a node.
+            outline: str. Free-form annotations that a lesson implementer
+                can use to construct the exploration. It describes the basic
+                theme or template of the story.
+            exploration_id: str or None. The valid exploration id that fits the
+                story node. It can be None initially, when the story creator
+                has just created a story with the basic storyline (by providing
+                notes) without linking an exploration to any node.
+        """
+        self.id = node_id
+        self.destination_node_ids = destination_node_ids
+        self.acquired_skill_ids = acquired_skill_ids
+        self.prerequisite_skill_ids = prerequisite_skill_ids
+        self.outline = outline
+        self.exploration_id = exploration_id
+
+    def to_dict(self):
+        """Returns a dict representing this StoryNode domain object.
+
+        Returns:
+            A dict, mapping all fields of StoryNode instance.
+        """
+        return {
+            'id': self.id,
+            'destination_node_ids': self.destination_node_ids,
+            'acquired_skill_ids': self.acquired_skill_ids,
+            'prerequisite_skill_ids': self.prerequisite_skill_ids,
+            'outline': self.outline,
+            'exploration_id': self.exploration_id
+        }
+
+    @classmethod
+    def from_dict(cls, node_dict):
+        """Return a StoryNode domain object from a dict.
+
+        Args:
+            node_dict: dict. The dict representation of StoryNode object.
+
+        Returns:
+            StoryNode. The corresponding StoryNode domain object.
+        """
+        node = cls(
+            node_dict['id'], node_dict['destination_node_ids'],
+            node_dict['acquired_skill_ids'],
+            node_dict['prerequisite_skill_ids'], node_dict['outline'],
+            node_dict['exploration_id'])
+
+        return node
+
+    @classmethod
+    def create_default_story_node(cls, node_id):
+        """Returns a StoryNode domain object with default values.
+
+        Args:
+            node_id: str. The id of the node.
+
+        Returns:
+            StoryNode. The StoryNode domain object with default
+            value.
+        """
+        return cls(node_id, [], [], [], '', '')
```

Comment:
shouldn't the default for exploration_id be None?

Btw, here and for the other domain objects, where's the validate() function? It should check things like this (e.g. exploration_id should not be an empty string).